From 8bfcbdb173cf4944cf996ac91566a8ff22e5b748 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Thu, 5 Jul 2018 20:18:29 +0200
Subject: [PATCH 1/1] efi_loader: correct signature of CalculateCrc32()

Use efi_uintn_t where prescribed by the UEFI spec.
Prefer u32 over uint32_t.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 include/efi_api.h             | 2 +-
 lib/efi_loader/efi_boottime.c | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/efi_api.h b/include/efi_api.h
index c98cc34908..0ab0cab469 100644
--- a/include/efi_api.h
+++ b/include/efi_api.h
@@ -166,7 +166,7 @@ struct efi_boot_services {
 	efi_status_t (EFIAPI *uninstall_multiple_protocol_interfaces)(
 			void *handle, ...);
 	efi_status_t (EFIAPI *calculate_crc32)(void *data,
-			unsigned long data_size, uint32_t *crc32);
+			efi_uintn_t data_size, u32 *crc32);
 	void (EFIAPI *copy_mem)(void *destination, const void *source,
 			size_t length);
 	void (EFIAPI *set_mem)(void *buffer, size_t size, uint8_t value);
diff --git a/lib/efi_loader/efi_boottime.c b/lib/efi_loader/efi_boottime.c
index 9ac8e18680..c5d23c81bc 100644
--- a/lib/efi_loader/efi_boottime.c
+++ b/lib/efi_loader/efi_boottime.c
@@ -2419,8 +2419,8 @@ static efi_status_t EFIAPI efi_uninstall_multiple_protocol_interfaces(
  * Return Value:	status code
  */
 static efi_status_t EFIAPI efi_calculate_crc32(void *data,
-					       unsigned long data_size,
-					       uint32_t *crc32_p)
+					       efi_uintn_t data_size,
+					       u32 *crc32_p)
 {
 	EFI_ENTRY("%p, %ld", data, data_size);
 	*crc32_p = crc32(0, data, data_size);
-- 
2.18.0

